core/templates/pages/preferences-page/preferences-page.component.ts

```diff
@@ -175,18 +191,53 @@ export class PreferencesPageComponent {
     }
   }
 
+  private saveProfileImage(image: Blob): void {
+    if (
+      this.contextService.getImageSaveDestination() ===
+      AppConstants.IMAGE_SAVE_DESTINATION_LOCAL_STORAGE
+    ) {
+      this._saveProfileImageToLocalStorage(image);
+    } else {
+      this._postProfileImageToServer(image);
+    }
+  }
+
+  private _saveProfileImageToLocalStorage(image: Blob): void {
+    const reader = new FileReader();
+    reader.onload = () => {
+      const imageData = reader.result as string;
+      this.imageLocalStorageService.saveImage(
+        'profile_image.png', imageData);
```

Comment:
NOTE: When running the dev server locally, the user's profile image is uploaded as `profile_image.png`.